CHANGELOG.md

```diff
@@ -6,6 +6,31 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 
 ## Unreleased
 
+## [0.31.0] - 2024-09-03
+
+### Added
+- Parameter `meta: dict` on `BaseEvent`.
+
+### Changed
+- **BREAKING**: Drivers, Loaders, and Engines will now raises exceptions rather than returning `ErrorArtifacts`.
+- **BREAKING**: Parameter `driver` on `BaseConversationMemory` renamed to `conversation_memory_driver`.
+- **BREAKING**: `BaseConversationMemory.add_to_prompt_stack` now takes a `prompt_driver` parameter.
+- **BREAKING**: `BaseConversationMemoryDriver.load` now returns `tuple[list[Run], Optional[dict]]`.
+- **BREAKING**: `BaseConversationMemoryDriver.store` now takes `runs: list[Run]` and `metadata: Optional[dict]` as input.
+- **BREAKING**: Parameter `file_path` on `LocalConversationMemoryDriver` renamed to `persist_file` and is now type `Optional[str]`.
+- `Defaults.drivers_config.conversation_memory_driver` now defaults to `LocalConversationMemoryDriver` instead of `None`.
+- `CsvRowArtifact.to_text()` now includes the header.
```

Comment:
is this always, or is it optional? Is there any case where someone would NOT want it?

Reply:
> Is there any case where someone would NOT want it?

I don't think so, but this highlights a fundamental issue with `CsvRowArtifact` that is being changed in 32.